sdk/sync/client_sugar.go

```diff
@@ -0,0 +1,66 @@
+package sync
+
+import "context"
+
+// PublishAndWait composes Publish and a Barrier. It first publishes the
+// provided payload to the specified topic, then awaits for a barrier on the
+// supplied state to reach the indicated target.
+//
+// If any operation fails, PublishAndWait short-circuits and returns a non-nil
+// error and a negative sequence. If Publish succeeds, but the Barrier fails,
+// the seq number will be greater than zero.
+func (c *Client) PublishAndWait(ctx context.Context, topic *Topic, payload interface{}, state State, target int) (seq int64, err error) {
```

Comment:
These are some "sugar" methods, which compose primitives into popular katas, for easier use.